### PR TITLE
Cursed scroll of remove curse

### PR DESCRIPTION
**Pull request by mschober**
test passes cursing dagger when casting cursed remove curse scroll
  pass in all the dwFlags

### CODE REVIEW

**Comment by mschober** [comment]
image

**Comment by mschober** [comment]
image

**Comment by mschober** [comment]
image

**Comment by mschober** [comment]
image

File: src/Player.cpp [hunk 1]
```diff
@@ -564,11 +564,11 @@ JResult CPlayer::Magic( CLink<CItem> *pLink )
     // Quaff and Magic are similar but magic is multi-use
     CItem *pItem = pLink->m_lpData;
     CLink<CEffect> *plEffect = pItem->m_id->m_llEffects->GetHead();
-    JResult retval = DoEffects( plEffect );
+    JResult retval = DoEffects( plEffect, pItem->m_dwFlags );
     return retval;
 }
 
-JResult CPlayer::DoEffects( CLink<CEffect> *plEffect )
+JResult CPlayer::DoEffects( CLink<CEffect> *plEffect, int dwFlags )
```
Comment (Rushwind13) [inline src/Player.cpp]: Please rename second parameter to dwItemFlags, to distinguish from the ones in the Effect

File: src/Player.cpp [hunk 2]
```diff
@@ -591,7 +591,7 @@ JResult CPlayer::DoEffects( CLink<CEffect> *plEffect )
             break;
         case EFFECT_TYPE_DESTROY:
             JLog( LOG_LEVEL_ERROR, true, "Destroying\n" );
-            DoDestroyEffects( pEffect );
+            DoDestroyEffects( pEffect, dwFlags );
```
Comment (Rushwind13) [inline src/Player.cpp]: Dwitemflags

File: src/Player.cpp [hunk 3]
```diff
@@ -711,14 +711,21 @@ JResult CPlayer::DoLightArea()
     return JBOGUSKEY;
 }
 
-JResult CPlayer::DoDestroyEffects( CEffect *pEffect )
+JResult CPlayer::DoDestroyEffects( CEffect *pEffect, int dwFlags )
```
Comment (Rushwind13) [inline src/Player.cpp]: Same

File: src/Player.cpp [hunk 4]
```diff
@@ -742,6 +750,26 @@ JResult CPlayer::DoRemoveCurse()
     return JSUCCESS;
 }
 
+JResult CPlayer::DoApplyCurse()
+{
+    CItem *cursed;
+    CLink<CItem> *pLink = m_llEquipment->GetHead();
+    while( pLink != NULL )
+    {
+        if( pLink->m_lpData->m_dwFlags | ITEM_FLAG_CURSED )
```
Comment (Rushwind13) [inline src/Player.cpp]: This bitwise logic is incorrect I think, you want `flags & mask == mask` for Boolean `is set`, or  `flags & mask == 0` for Boolean `is not set`

File: src/Player.h [hunk 5]
```diff
@@ -175,15 +175,16 @@ class CPlayer
     float LightSource();
     void UpdateLight( float fValue, bool bReset = false );
 
-    JResult DoEffects( CLink<CEffect> *plEffect );
+    JResult DoEffects( CLink<CEffect> *plEffect, int dwFlags );
```
Comment (Rushwind13) [inline src/Player.h]: dwItemFlags on these declarations

**Review by Rushwind13 — CHANGES_REQUESTED** [review]
Quick rename on the flag variable, and check your Boolean logic in the new function.

File: src/Player.cpp (hunk 4 above)
Comment (mschober) [reply src/Player.cpp]: I see
```
  101100 - some current flag state
& 001000 - some mask
=======
   001000 - yields the mask
```

**Review by Rushwind13 — APPROVED** [review]

**Pull request merged**